frontend/src/static/js/components/webstatus-line-chart-panel.ts

```diff
@@ -0,0 +1,342 @@
+/**
+ * Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {CSSResultGroup, LitElement, TemplateResult, css, html} from 'lit';
+import {property, state} from 'lit/decorators.js';
+import {WebStatusDataObj} from './webstatus-gchart.js';
+import {Task} from '@lit/task';
+import {APIClient, apiClientContext} from '../contexts/api-client-context.js';
+import {consume} from '@lit/context';
+import {SHARED_STYLES} from '../css/shared-css.js';
+
+/**
+ * Interface defining the structure of metric data for the line chart.
+ * @template T The type of the data items.
+ */
+export interface LineChartMetricData<T> {
+  /**
+   * The label for the metric (displayed on the chart legend).
+   * @type {string}
+   */
+  label: string;
+
+  /**
+   * The array of data points for the metric.
+   * @type {Array<T>}
+   */
+  data: Array<T>;
+
+  /**
+   * Function to extract the timestamp from a data item.
+   * @param {T} item The data item.
+   * @returns {Date} The timestamp of the data item.
+   */
+  getTimestamp: (item: T) => Date;
+
+  /**
+   * Function to extract the data value from a data item.
+   * @param {T} item The data item.
+   * @returns {number | undefined} The data value of the data item.
+   */
+  getData: (item: T) => number | undefined;
```

Comment:
Maybe `getValue` would be more clear since the word "data" can mean a single value or a whole list.

Reply:
Thanks! Done!